tests/jballerina-unit-test/src/test/resources/test-src/types/table/record-type-table-key.bal

```diff
@@ -193,12 +193,52 @@ function testInferTableTypeV2() {
 const int id = 1;
 
 function testVariableNameFieldAsKeyField() {
-    string expected = "[{\"id\":1,\"name\":\"Jo\"},{\"id\":2,\"name\":\"Amy\"}]";
     table<record {readonly int id; string name;}> key (id) tb = table [
         {id, name: "Jo"},
         {id: 2, name: "Amy"}
     ];
-    assertEquality(expected, tb.toString());
+    assertEquality(tb[1], <record {readonly int id; string name;}> {"id": 1, "name": "Jo"});
+    assertEquality(tb[2], <record {readonly int id; string name;}> {"id": 2, "name": "Amy"});
+}
+
```

Comment:
Let's add tests for when the table constructor specifies the key specifier also.

E.g., `table<Employee> e = table key(id) [...]`.

Reply:
Currently, Ballerina does not member access for keyless table

Reply:
You can use an iterator to iterate through the members.

Reply:
This code is not supported in current version

```
import ballerina/io;

const int id = 1;

public function main() {
    record {readonly int id; string name;}[] expected = [
        <record {readonly int id; string name;}> {"id": 1, "name": "Jo"},
        <record {readonly int id; string name;}> {"id": 2, "name": "Amy"}
    ];
    table<record {readonly int id; string name;}> tb = table key (id) [
        {id, name: "Jo"},
        {id: 2, name: "Amy"}
    ];

    foreach int i in 0..< tb.length() {
        io:println(tb[i + 1], expected[i]);
    }
}
```

Reply:
`ERROR [app.bal:(16:20,16:29)] member access is not supported for keyless table 'tb`

Reply:
Did you try using an iterator on the table as suggested?

Reply:
Updated